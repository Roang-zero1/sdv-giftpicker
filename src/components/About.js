import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from 'reactstrap';

/* TODO: Add an about referencing the used tools*/

class About extends Component {
  render() {
    return (
      <Container className="about">
        <h2>About</h2>
        <p>
          SDV Gift Calculator by{' '}
          <a href="https://www.reddit.com/user/Roang_zero1/">Roang_Zero1</a>.
        </p>
        <p>
          This app was inspired by <strong>Stardew Checkup</strong> available at{' '}
          <a href="https://mouseypounds.github.io/stardew-checkup/">
            https://mouseypounds.github.io/stardew-checkup/
          </a>
        </p>
      </Container>
    );
  }
}

function mapStateToProps(state) {
  return {
    status: state.status
  };
}

export default connect(mapStateToProps)(About);
