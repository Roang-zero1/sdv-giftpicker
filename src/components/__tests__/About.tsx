import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import About from '../About';

describe('components/About --- Shallow render component', () => {
  let cut: ShallowWrapper;

  beforeEach(() => {
    cut = shallow(<About />);
  });

  it('should render the one component', () => {
    expect(cut.length).toEqual(1);
  });

  it('should be the same as the last snapshot', () => {
    expect(toJson(cut)).toMatchSnapshot();
  });
});
