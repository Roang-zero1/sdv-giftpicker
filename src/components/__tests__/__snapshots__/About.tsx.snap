// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/About --- Shallow render component should be the same as the last snapshot 1`] = `
<Col
  className="about"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
  xs="12"
>
  <h2>
    About
  </h2>
  <p>
    SDV Gift Calculator by
     
    <a
      href="https://www.reddit.com/user/Roang_zero1/"
    >
      Roang_Zero1
    </a>
    .
  </p>
  <p>
    This app was inspired by 
    <strong>
      Stardew Checkup
    </strong>
     available at
     
    <a
      href="https://mouseypounds.github.io/stardew-checkup/"
    >
      https://mouseypounds.github.io/stardew-checkup/
    </a>
  </p>
  <p>
    Using icons by 
    <a
      href="https://fontawesome.com/l"
    >
      Font Awesome
    </a>
     
    licensed under
     
    <a
      href="https://creativecommons.org/licenses/by/4.0/"
    >
      CC BY 4.0
    </a>
  </p>
</Col>
`;
